docs/tutorial/Tutorial.md

```diff
@@ -381,31 +419,27 @@ and prepare the new spaces, and a copy context.
 
 Firstly, change the plan constraints. Some of these constraints are not used 
 at the moment, but it's good to set them properly regardless.
-1. Look in `plan/plan_constraints.rs`. This file contains all the possible 
-options for constraints. At the moment, `mygc/constraints.rs` contains 2 
-variables: `GC_HEADER_BITS` and `GC_HEADER_WORDS`. Both are set to 0. Change 
-`GC_HEADER_BITS` to 2.
-2. You will need to change two more options: `MOVES_OBJECTS` and 
-`NUM_SPECIALIZED_SCANS`. Copy the lines containing both to 
-`mygc/constraints.rs`.
-3. Set `MOVES_OBJECTS` to `true`. 
-4. Set `NUM_SPECIALIZED_SCANS` to 1.
-
-[[Finished code (step 1-4]](/docs/tutorial/tutorial%20code/mygc_semispace/constraints.rs#L1-L6)
-
-Next, in `global.rs`, replace the old immortal space with two copyspaces.
+1. Look in `plan/plan_constraints.rs`. `PlanConstraints` lists all the possible
+options for plan-specific constraints. At the moment, `MYGC_CONSTRAINTS` in `mygc/global.rs` should be using
+the default value for `PlanConstraints`. We will make the following changes.
+1. Initialize `gc_header_bits` to 2. We reserve 2 bits in the header for GC use.
+1. Initialize `moves_objects` to `true`.
+1. Initialize `num_specialized_scans` to 1.
+[[Finished code (step 1-4]](/docs/tutorial/code/mygc_semispace/global.rs)
+
+Next, in `global.rs`, replace the old immortal (nogc) space with two copyspaces.
 1. To the import statement block:
    1. Replace `crate::plan::global::{BasePlan, NoCopy};` with 
    `use crate::plan::global::BasePlan;`. This collector is going to use 
    copying, so there's no point to importing NoCopy anymore.
-   2. Add `use crate::plan::global::CommonPlan;`. semispace uses the common 
+   2. Add `use crate::plan::global::CommonPlan;`. Semispace uses the common
    plan, which includes an immortal space and a large object space, rather 
    than the base plan. Any garbage collected plan should use `CommonPlan`.
    3. Add `use std::sync::atomic::{AtomicBool, Ordering};`. These are going 
    to be used to store an indicator of which copyspace is the tospace.
    4. Delete `#[allow(unused_imports)]`.
    
-   [[Finished code (step 1)]](/docs/tutorial/tutorial%20code/mygc_semispace/global.rs#L2-L28)
+   [[Finished code (step 1)]](/docs/tutorial/code/mygc_semispace/global.rs)
```

Comment:
The line numbers are missing here, and subsequent [[Finished code..]]

Reply:
As described in the PR description, I removed all the line numbers as they are hard to maintain correctly. If we need to update the tutorial code, we need to manually go through all the line numbers and fix them one by one. It is expensive. Let me know if you suggest keeping the line number.

Reply:
I agree that it's expensive, but I think the links will be much less useful without them.
Overall, my suggestion is that we keep them

Reply:
I added back the line numbers.